File: my-frontend/src/types/series.ts
import { ArticleDto } from "./article";

export interface SeriesDto {
  id?: string; // cho phép undefined
  title?: string;
  slug?: string;
  description?: string;
  thumbnailUrl?: string;
  authorName?: string;
  articles?: ArticleDto[];
}

export interface SeriesSummaryDto {
  id?: string; // cho phép undefined
  title?: string;
  slug?: string;
  description?: string;
  thumbnailUrl?: string;
}

File: QuizAppBackend/src/main/java/com/kkunquizapp/QuizAppBackend/article/model/Series.java
package com.kkunquizapp.QuizAppBackend.article.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.GenericGenerator;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;
import java.util.UUID;

@Entity
@Table(name = "series")
@Getter
@Setter
public class Series {

    @Id
    @GeneratedValue(generator = "uuid2")
    @GenericGenerator(name = "uuid2", strategy = "org.hibernate.id.UUIDGenerator")
    @Column(columnDefinition = "UUID")
    private UUID id;

    @Column(nullable = false)
    private String title;

    @Column(unique = true, nullable = false)
    private String slug;

    @Column(columnDefinition = "TEXT")
    private String description;

    private String thumbnailUrl;

    @Column(nullable = false)
    private UUID authorId;

    @OneToMany(fetch = FetchType.LAZY, mappedBy = "series")
    private Set<ArticleSeries> articles = new HashSet<>();

    private LocalDateTime createdAt = LocalDateTime.now();
    private LocalDateTime updatedAt = LocalDateTime.now();
}

